components/src/core/ToolbarMenu/ToolbarMenu.tsx

```diff
@@ -0,0 +1,258 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import ArrowDropDown from '@material-ui/icons/ArrowDropDown';
+import clsx from 'clsx';
+import createStyles from '@material-ui/core/styles/createStyles';
+import composeRefs from '@seznam/compose-react-refs';
+import { DrawerNavGroup, NavItem } from '../Drawer';
+import makeStyles from '@material-ui/core/styles/makeStyles';
+import Menu, { MenuProps as standardMenuProps } from '@material-ui/core/Menu';
+import PropTypes from 'prop-types';
+import { Theme } from '@material-ui/core/styles/createMuiTheme';
+import useTheme from '@material-ui/core/styles/useTheme';
+import Typography, { TypographyProps } from '@material-ui/core/Typography';
+
+export type ToolbarMenuClasses = {
+    root?: string;
+    dropdownArrow?: string;
+    icon?: string;
+    label?: string;
+};
+
+export type ToolbarMenuCompItem = Omit<NavItem, 'itemID'> & { itemID?: string };
+export type ToolbarMenuCompGroup = {
+    /** The color used for the text */
+    fontColor?: string;
+    /** The color used for icons */
+    iconColor?: string;
+    /** List of navigation items to render */
+    items: ToolbarMenuCompItem[];
+    /** Text to display in the group header */
+    title?: string;
+};
+
+const useStyles = makeStyles((theme: Theme) =>
+    createStyles({
+        root: {
+            cursor: 'pointer',
+            display: 'flex',
+            alignItems: 'center',
+            maxWidth: 'fit-content',
+        },
+        cursorPointer: {
+            cursor: 'pointer',
+        },
+        dropdownArrow: {
+            marginLeft: theme.spacing(0.5),
+        },
+        icon: {
+            marginRight: `${theme.spacing(1)}px`,
+            display: 'inline-flex',
+            fontSize: 'inherit',
+        },
+        label: {
+            textOverflow: 'ellipsis',
+            overflow: 'hidden',
+        },
+        navGroups: {
+            '&:active, &:focus': {
+                outline: 'none',
+            },
+        },
+        rotateDropdownArrow: {
+            transform: 'rotate(180deg)',
+        },
+    })
+);
+
+export type ToolbarMenuProps = TypographyProps & {
```

Comment:
May want to omit the onClick here since we're overriding it and not merging with anything the user provides:
```suggestion
export type ToolbarMenuProps = Omit<TypographyProps, 'onClick'> & {
```